### PR TITLE
Add typescript jest

### PR DESCRIPTION
**Pull request by ReaSu**

### CODE REVIEW

File: typescript/src/SimpleEnemy.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {Target} from './Target';
+import {Armor} from './Armor';
+import {Buff} from './Buff';
+
```
Comment (codecop) [inline typescript/src/SimpleEnemy.ts:4]: two empty lines

File: typescript/tests/MockPlayer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+});
+
+jest.mock('../src/SimpleEnemy', () => {
+    return {
+        SimpleEnemy: jest.fn().mockImplementation(() => {
+            return {/* TODO */};
+        }),
+    };
+});
+
```
Comment (codecop) [inline typescript/tests/MockPlayer.spec.ts]: haven't used Jest in this way. I have used
```
    hardware = {
      read: jest.fn((a) => 0),
      write: jest.fn((a, v) => { })
    };

```
for mocks, i.e. mocked each method. Was simpler. Maybe your code is the way it is intended

File: typescript/tests/Player.spec.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { Inventory } from '../src/Inventory';
+import { Stats } from '../src/Stats';
+import { SimpleEnemy } from '../src/SimpleEnemy';
+import { Player } from '../src/Player';
+import { Damage } from '../src/Damage';
+
+// use this test if you are not familiar with mocks
```
Comment (codecop) [inline typescript/tests/Player.spec.ts]: move this line to line 1 as in previous test

**Review by codecop — APPROVED** [review]

**Comment by codecop** [comment]
Also we could have less tests. Swift and Python have no tests. But as you created them. I guess they are good. maybe mocking is a bit verbose...

**Comment by codecop** [comment]
Also the other languages do not have the extra Readme. Do you think it is necessary, then all languages should have it.

File: typescript/tests/MockPlayer.spec.ts (hunk 2 above)
Comment (ReaSu) [reply typescript/tests/MockPlayer.spec.ts]: I tried mocking the functions only. But I need mock objects of Inventory and Stats to create the OUT, i.e. the player.
Or how would you do that?

**Comment by ReaSu** [comment]
Maybe having no tests is the best solution. :·)

I added the special Readme because 

1. The Parrot Kata had one, saying that it is an npm project etc.
2. The other languages only have one test file, so it's easier to spot that there are two ways to test. 

But if we leave out the tests, then I don't think we'd need the extra Readme.

I didn't find any forks with Typescript, btw.

File: typescript/tests/MockPlayer.spec.ts (hunk 2 above)
Comment (ReaSu) [reply typescript/tests/MockPlayer.spec.ts]: I used this part of the documentation: https://jestjs.io/docs/es6-class-mocks

**Comment by codecop** [comment]
Let's maybe keep the simpler test without mocks. CPP has that. It's a starting point nevertheless

**Comment by ReaSu** [comment]
That's a good compromise, I like it. I removed the Readme as well.

**Pull request merged**